src/_pytest/fixtures.py

```diff
@@ -936,6 +956,44 @@ def _ensure_immutable_ids(ids):
     return tuple(ids)
 
 
+def wrap_function_to_warning_if_called_directly(function, fixture_marker):
+    """Wrap the given fixture function so we can issue warnings about it being called directly, instead of
+    used as an argument in a test function.
+
+    The warning is emitted only in Python 3, because I didn't find a reliable way to make the wrapper function
+    keep the original signature, and we probably will drop Python 2 in Pytest 4 anyway.
+    """
+    if six.PY2:
+        return function
+
+    is_yield_function = is_generator(function)
+    msg = FIXTURE_FUNCTION_CALL.format(name=fixture_marker.name or function.__name__)
+    warning = RemovedInPytest4Warning(msg)
+
+    if is_yield_function:
+
+        @functools.wraps(function)
```

Comment:
functools.wraps is broken on python2 and needs to set `__wrapped__` or so to get the correct metadata